ding/worker/collector/base_serial_evaluator.py

```diff
@@ -127,6 +128,41 @@ def update_reward(self, env_id: int, reward: Any) -> None:
             reward = reward.item()
         self._reward[env_id].append(reward)
 
+    def update_video(self, imgs):
+        for env_id, img in imgs.items():
+            if len(self._reward[env_id]) == len(self._video[env_id]):
+                self._video[env_id].append([])
+            self._video[env_id][-1].append(img)
+
+    def get_video(self):
+        """
+        Overview:
+            Convert list of videos into [N, T, C, H, W] tensor, containing
+            worst, median, best evaluation trajectories for video logging.
+        """
+        videos = sum([list(v)[:e] for e, v in zip(self._each_env_episode, self._video.values())], [])
+        videos = [np.transpose(np.stack(video, 0), [0, 3, 1, 2]) for video in videos]
+        sortarg = np.argsort(self.get_episode_reward())
+        # worst, median(s), best
+        if len(sortarg) == 1:
+            idxs = [sortarg[0]]
+        elif len(sortarg) == 2:
+            idxs = [sortarg[0], sortarg[-1]]
+        elif len(sortarg) == 3:
+            idxs = [sortarg[0], sortarg[len(sortarg) // 2], sortarg[-1]]
+        else:
+            idxs = [sortarg[0], sortarg[len(sortarg) // 2 - 1], sortarg[len(sortarg) // 2], sortarg[-1]]
+        videos = [videos[idx] for idx in idxs]
+        # pad videos to the same length
```

Comment:
why not use `np.pad` here

Reply:
I pad each video with the its last frame instead of a constant value, it is not obvious for me how to implement it with `np.pad` and the current implementation is not hard to read anyway.